cli/command/container/opts.go

```diff
@@ -335,12 +336,29 @@ func parse(flags *pflag.FlagSet, copts *containerOptions) (*containerConfig, err
 	for bind := range copts.volumes.GetMap() {
 		parsed, _ := loader.ParseVolume(bind)
 		if parsed.Source != "" {
-			// after creating the bind mount we want to delete it from the copts.volumes values because
-			// we do not want bind mounts being committed to image configs
-			binds = append(binds, bind)
 			// We should delete from the map (`volumes`) here, as deleting from copts.volumes will not work if
 			// there are duplicates entries.
 			delete(volumes, bind)
+
+			if parsed.Type == string(mount.TypeBind) {
+				// Resolve relative path of source volume if needed
+				resolved, err := resolveLocalPath(parsed.Source)
+				if err != nil {
+					return nil, err
+				}
+				// use the resolved file path
+				bind = resolved + ":" + parsed.Target
+				if parsed.ReadOnly {
+					bind += ":ro"
+				}
```

Comment:
Based on the failing test, I think maybe you need to append `":rw"` in an `else` here?

Reply:
It does seem like it is! I’ll do the change and test it shortly :) thanks for the suggestion and attention to this PR.